### ISSUE TITLE
No Extraction Button

### ISSUE THREAD

**Issue by smilerz** [joonaspaakko/audible-library-extractor#151]
I've previously used the extension successfully, however at the moment I don't see the extraction button below search on the library page and manually selecting the extension in the browser doesn't work either.

I'm using Edge 130.0.2849.13

I think the console is showing the extension being blocked here:

`content-script-loader.content-script.ec187947.01f7d99a.js:7  Refused to load the script 'chrome-extension://<guid>/assets/content-script.ec187947.js' because it violates the following Content Security Policy directive: "script-src 'self' 'wasm-unsafe-eval' 'inline-speculation-rules'". Note that 'script-src-elem' was not explicitly set, so 'script-src' is used as a fallback.
`

**Comment by joonaspaakko** [comment]
The filename should be "audible-library-extractor-content-script.js" so I don't think that's related. There have been a couple issues where Audible website changed and the extension couldn't find where to put the button anymore, though I tested it briefly and it was there. However, I did it such a haste that I'm not sure which build I tried, it might've been a dev build. I'll test it properly later.

If there are no errors thst prevent it from working, to going to your library and then click the extension icon in the extension toolbar. Thst should also open the extraction settings.

**Comment by smilerz** [comment]
I tried the extension icon and it doesn't perform any action - and doesn't output anything to the browser console.

I'm not certain why the naming is what it is - but that script mentioned above is related to this extension.  Opening the script manually you see things like this

`>Performance improvement <a href='https://github.com/joonaspaakko/audible-library-extractor/issues/41'>#41</a> changes the way the books are loaded in: they are loaded in chunks as you scroll down.`

**Comment by joonaspaakko** [comment]
Is it possible you have an older version installed? The newest [store version is 0.2.10](https://chromewebstore.google.com/detail/audible-library-extractor/deifcolkciolkllaikijldnjeloeaall)?

**Comment by smilerz** [comment]
That is the version I have installed.  I've also attempted to remove and reinstall with the same behavior.

The extension works in Chrome - so I do wonder if it is something that Edge has done.

**Comment by joonaspaakko** [comment]
Right, I tested it in Chrome. It is the same exact build, so the only difference is the browser and its possible Edge had an update that affected extension policies/rules, since that error message has to do with what the extension is allowed to do. I'll look into it a bit more later.

**Comment by wizardfish2** [comment]
Buggers.  I seem to have gotten hit with this same (no extraction button) issue over the past 3 days.  I was last able to do an extract/update on 10/19, but today I can no longer update.  Tried on both Chrome (which updated yesterday I believe, to version "Version 130.0.6723.59 (Official Build) (64-bit)"), and also on Edge on the same machine (Version 130.0.2849.46 (Official build) (64-bit), not sure when last updated but I don't use Edge for extraction so might not have noticed it change).

Also tried on multiple machines here, same issue.  (Chrome Version 130.0.6723.60 (Official Build) (64-bit)).  

Fingers crossed it's either temporary or something that is easily fixable.  :-)

**Comment by wizardfish2** [comment]
Update:  seems like chrome thinks something is unsafe in the extension.  (Looks like the same message as the first in this thread.)

content-script-loade…87947.01f7d99a.js:7 Refused to load the script 'chrome-extension://15fce5b4-d008-4094-8240-6adda47fa057/assets/content-script.ec187947.js' because it violates the following Content Security Policy directive: "script-src 'self' 'wasm-unsafe-eval' 'inline-speculation-rules'". Note that 'script-src-elem' was not explicitly set, so 'script-src' is used as a fallback.

If I dig into the link, it definitely is the audible-library-extractor, as there are many references to it in the referenced .js file.

Darn you, Google!  :-(

**Comment by joonaspaakko** [comment]
I have a strong feeling you could get around it temporarily if you install an older version of the browser. If you don't want to mess with the browser you use regularly, then perhaps an older version of some other Chromium based browser.

**Comment by joonaspaakko** [comment]
As I suspected, I had Chrome Version `129`  and when I updated to `130` I got the same issue, which seems to match with this issue post for the build tool I'm using https://github.com/crxjs/chrome-extension-tools/issues/918

So any Chromium based browser that uses anything older than 130 should probably work until this gets fixed.

Some options if you want to keep your main browser up to date:
- **Brave:** 
    - Last Chromium v.129 of [Brave browser](https://github.com/brave/brave-browser/releases/tag/v1.70.123). 
- **Vivaldi:**
    - Couldn't find a smart way to link to a specific version for all platforms, so [here you can find all old versions](https://vivaldi.com/fi/download/archive/) and the first one that I'm seeing in the list "v.6.9" seems to use an older Chromium version.

**Comment by pilgrimshoes** [comment]
Thanks very much, that worked for me.

I only run ALE once or twice a year. Something that worked well for my needs, in case it's useful to others:

- download the zip package of Brave (no installer) and extract files
- run Brave from the exe
- add the ALE extension and use it

The Extraction button appeared in my Library, and ALE worked beautifully. I expect it will work until Audible.com breaks something.

Maybe of interest to those who, if possible, would rather not install / uninstall software; or install and provide access to extensions within their main browser.

>     * **Brave:**
>       
>       * [Last Chromium v.129 ](https://github.com/brave/brave-browser/releases/tag/v1.70.123).
>

**Comment by eserafina42** [comment]
I'm having the same issue - very frustrating since I have a ton of books and Audible provides no easy way to filter the ones from the Plus subscription or let you find out which ones are expired, which is mainly what I use it for. I hope you (or the browsers) will find a way around this at some point.

**Comment by joonaspaakko** [comment]
@smilerz, @wizardfish2, @pilgrimshoes, @eserafina42 

I believe all the major bugs have been addressed, and it should work in any new Chromium versions. I'm still seeing an unusual amount of errors in the `chrome:extensions` error log, but it seems to function, not that I've tested every single feature yet (mainly CSV export and wallpaper creator).  

I figured that I'd let you know in case you want to try it before I do the release, which now that I'm back to work may easily take a week.

> **Draft installation instructions**
> 1. [Download the current v.0.2.11 draft](https://github.com/joonaspaakko/audible-library-extractor/raw/refs/heads/master/audible-library-extractor-0.2.11.zip).
> 2. Go to the address: `chrome:extensions` or: `Window > Extensions`.
> 3. Turn on the developer mode from the top right
> 4. Drag the downloaded .zip file in the browser window to install

**Comment by smilerz** [comment]
Good timing!  I was just looking to do an extraction.

For my use case it seems to work fine, thanks!
